src/gui/editors/AutomationEditor.cpp

```diff
@@ -615,6 +615,91 @@ void AutomationEditor::mousePressEvent( QMouseEvent* mouseEvent )
 
 
 
+void AutomationEditor::mouseDoubleClickEvent(QMouseEvent * mouseEvent)
+{
+	if (!validPattern()) { return; }
+
+	// Helper lambda function to fine tune a node's in/out value
+	auto fineTuneValue = [this](timeMap::iterator node, bool editingInValue)
+	{
+		if (node != m_pattern->getTimeMap().end())
+		{
+			// Display dialog to edit the value
+			bool ok;
+			double value = QInputDialog::getDouble(
+				this,
+				tr("Edit Value"),
+				editingInValue
+					? tr("New inValue")
+					: tr("New outValue"),
+				editingInValue
+					? INVAL(node)
+					: OUTVAL(node),
+				m_pattern->firstObject()->minValue<float>(),
+				m_pattern->firstObject()->maxValue<float>(),
+				3,
+				&ok
+			);
+
+			if (ok)
+			{
+				// Set the new inValue/outValue
+				if (editingInValue)
+				{
+					// If the outValue is equal to the inValue we
+					// set both to the given value
+					if (OFFSET(node) == 0)
+					{
+						node.value().setOutValue(value);
+					}
+					node.value().setInValue(value);
+					Engine::getSong()->setModified();
+				}
+				else
+				{
+					node.value().setOutValue(value);
+					Engine::getSong()->setModified();
+				}
+
+				return true;
+			}
+		}
+
+		// If nothing was changed (invalid node, dialog closed, etc) return false
+		return false;
+	};
+
+	// Do we need to call update()?
+	bool needUpdate = false;
```

Comment:
Any reason you don't just call ``update()`` in the places where you set ``needUpdate = true``? Otherwise, you could at least move this to be inside the if statement below since it's only used there.

Reply:
To be honest I don't remember, maybe back then I thought it was more readable or something. You prefer to replace it with a straight call to `update` or moving it inside the `if` block?

Reply:
The way the method looks now I prefer a straight call to update.